node/Dockerfile

```diff
@@ -0,0 +1,60 @@
+# Miden node Dockerfile
+
+# Setup image builder
+FROM rust:1.76-slim-bookworm AS builder
+
+# Install dependencies
+RUN apt-get update && \
+    apt-get -y upgrade && \
+    apt-get install -y llvm clang bindgen pkg-config libssl-dev libsqlite3-dev && \
+    rm -rf /var/lib/apt/lists/*
+
+# Copy source code
+WORKDIR /app
+COPY . .
+
+# Cache Cargo dependencies
+RUN --mount=type=cache,target=/usr/local/cargo/registry \
+    cargo fetch --manifest-path node/Cargo.toml
+
+# Build the node crate
+RUN cargo install --features testing --path node
+RUN miden-node make-genesis --inputs-path node/genesis.toml
+
+# Run Miden node
+FROM debian:bookworm-slim
+
+# Install required packages
+RUN apt-get update && \
+    apt-get install -y --no-install-recommends \
+    libssl-dev \
+    libsqlite3-dev \
```

Comment:
Do we need these dependencies? If so, can you please add some docs about it?

Reply:
This is the runner image, no?

Reply:
Yes it is indeed. Running the program failed when these were not installed on the machine.

Reply:
That is interesting. Can you please add a comment above the deps. It seems a bit off that the headers are needed to run the compiled binary. Also, what was the error?

Reply:
2024-03-12 16:55:51 miden-node: error while loading shared libraries: libsqlite3.so.0: cannot open shared object file: No such file or directory

Reply:
Ah, for that you shouldn't need the headers, you only need the library. `libsqlite3` should be enough, it is only marginally better though (to avoid confusion like I had here)

Reply:
Here is what I have been proposing:

- keep dependencies to a minimum
- document why they are needed

why am I proposing the above?

- it is hard to maintain script files without this information, for example, the discussion of `gcc` and `clang` being needed is something that happened because of lack of documentation. I'm trying to prevent that
- the above is specially a problem when upgrade the file down the line, and usually causes issues were one doesn't know why something is there, so we default to "better not touch it since it is working", which can compound over time

so here, since `libsqlite3-dev` is not needed, these are the headers, we would remove that dependency and have `libsqlite3` instead. since that is the `.so` file that is needed to run the software, and add a comment `# sqlite3: used by the store`.

That way we have a minimum number of dependencies and some traceability on them, making it easier to maintain it.